src/adaptix/_internal/morphing/facade/provider.py

```diff
@@ -368,3 +369,7 @@ def validating_loader(data):
         raise exception_factory(data)
 
     return loader(pred, validating_loader, chain)
+
+
+def default_dict(default_factory: Optional[Type] = None) -> Provider:
+    return DefaultDictProvider(default_factory)
```

Comment:
All high-level factories should require predicate. Using default_dict without predicate is pointless, because you cannot apply same default_factory to all default dicts.
You can take an example of using predicate from ``loader``  or ``dumper`` factory.